File: iPhone/Apex Cougars/Classes/SeasonsService.m
//
//  SeasonsService.m
//  iAthletics Framework
//
//  Created by Robert Signore on 6/13/10.
//  Copyright 2010 Appapro. All rights reserved.
//

#import "SeasonsService.h"


@implementation SeasonsService

@synthesize seasons;
@synthesize curSeason;


#pragma mark constructors/destructors

-(id) initWithApplication:(iAthleticsApplication *)theApp
{
	self = [super initWithApplication:theApp];
	
	NSMutableArray *newMut = [[NSMutableArray alloc]initWithCapacity:10];
	self.seasons = newMut;
	
	[newMut release];
	
	return self;
}

-(void)dealloc
{
	[self.seasons release];
	[self.curSeason release];
	
	[super dealloc];
}

-(NSArray *)requestService
{
	[self requestServiceName:@"GetAppSeasons" parameters:nil];
	
	return self.seasons;
}

#pragma mark -
#pragma mark iAthleticsService ABSTRACT methods


-(void) elementStart:(NSString *)elementName
{
	// prepare to receive new AppSeason Info
	//
	if([elementName compare:@"AppSeason"] == NSOrderedSame)
	{
		AppSeason *newSeason = [[AppSeason alloc]init];
		self.curSeason = newSeason;
		
		[newSeason release];
	}
}

-(void)elementEnd:(NSString *)elementName withText:(NSString *)text
{
	// is this the season's name
	//
	if([elementName compare:@"SeasonName"] == NSOrderedSame)
	{
		self.curSeason.seasonName = text;
	}
	
	// the season ID?
	//
	else if([elementName compare:@"SeasonId"] == NSOrderedSame)
	{
		self.curSeason.seasonId = [text intValue];
	}
	
	// the icon URL?
	//
	else if([elementName compare:@"IconUrl"] == NSOrderedSame)
	{
		iAthleticsImage *newImage = [[iAthleticsImage alloc]initWithUrl:text];
		self.curSeason.image = newImage;
		
		[newImage release];
	}
	
	// if this is the end of the AppSeason then add the curSeason to the 
	// seasons array
	else if([elementName compare:@"AppSeason"] == NSOrderedSame)
	{
		[self.seasons addObject:self.curSeason];
	}
}


@end

File: Android/iAthletics/src/com/appapro/iathletics/services/SeasonsService.java
package com.appapro.iathletics.services;

import java.util.ArrayList;

import com.appapro.iathletics.activities.iAthleticsActivity;
import com.appapro.iathletics.types.AppSeason;
import com.appapro.iathletics.types.iAthleticsImage;

public class SeasonsService extends iAthleticsService
{
	// constant for Api call
	//
	private static final String SERVICE_NAME = "GetAppSeasons";
	
	// constants for tags in the XML
	//
	private static final String APPSEASON_TAG = "AppSeason";
	private static final String SEASONNAME_TAG = "SeasonName";
	private static final String SEASONID_TAG = "SeasonId";
	private static final String ICONURL_TAG = "IconUrl";
	
	ArrayList<AppSeason> seasons;
	AppSeason curSeason = null;

	public SeasonsService(iAthleticsActivity activity)
	{
		super(activity);
		
		// allocate some space
		//
		seasons = new ArrayList<AppSeason>(3);

	}
	
	// call this function to get the season data for the organization
	//
	public ArrayList<AppSeason> getSeasons()
	{
		this.requestService(SeasonsService.SERVICE_NAME, null);
		
		return this.seasons;
	}

	@Override
	protected void elementStart(String elementName)
	{
		// prepare to receive new AppSeason Info
		//
		if(elementName.compareTo(SeasonsService.APPSEASON_TAG) == 0)
		{
			this.curSeason = new AppSeason();
		}
	}

	@Override
	protected void elementEnd(String elementName, String text)
	{
		// is this the season's name
		//
		if(elementName.compareTo(SeasonsService.SEASONNAME_TAG) == 0)
		{
			this.curSeason.setSeasonName(text);
		}
		
		// the season ID?
		//
		else if(elementName.compareTo(SeasonsService.SEASONID_TAG) == 0)
		{
			this.curSeason.setSeasonId(Integer.valueOf(text));
		}
		
		// the icon URL?
		//
		else if(elementName.compareTo(SeasonsService.ICONURL_TAG) == 0)
		{
			this.curSeason.setImage(new iAthleticsImage(text, getActivity()));
		}
		
		// if this is the end of the AppSeason then add the curSeason to the 
		// seasons array
		//
		else if(elementName.compareTo(SeasonsService.APPSEASON_TAG) == 0)
		{
			this.seasons.add(curSeason);
		}
	}

}
